### DISCUSSION TITLE
How to best report some particular ROMs not working?

### DISCUSSION THREAD

**Discussion by jimjamz** [86Box/86Box#4715] [Q&A]
I've found at least two ROMs that do not boot even with the default configuration:
- [1992] i486 (Socket 2) / [SIS 461] IBM PS/ValuePoint 433 DX/Si
- [1992] i486 (Socket 2) / [i420TX] J-Bond PCI400C-A

**Pre-requisite steps:**
0. Boot and save configuration from a different ROM from within the same socket category with success.
OR
0. Remove any pre-existing existing 86box.cfg configuration file.

**Reproduction steps:**
1.  Select _Tools_ -> _Settings_.
2. From the _Machine_ section, choose the _Machine Type_.
3. From the _Machine_ section, choose the _Machine_.
4.  Selecting or not selecting any further configuration has no bearing on the result.
5.  Save the configuration.
6.  Start/Restart the machine.

**EXPECTED**:
The machine's BIOS will POST.

**ACTUAL**:
The machine's BIOS does not POST or the 86Box display output for the machine does not render/ is frozen.

**Host OS:**
Debian 12 Bookworm

**Tested with:**
- 86Box v4.2.1 [x86_64, new dynarec] - local build with cmake
- 86Box-Linux-x86_64-b6000.AppImage v4.2.1

Is there an appropriate process to follow when reporting ROMs that do not work?

**Comment by eddmanx** [comment]
How does that even mean that the problem is with the ROM. They boot here, but not with the default video adapter, which is not ROM related anyway.

**Reply by jimjamz** [reply]
I didn't say that the problem is specific to the ROM. I described that it does not work in the default configuration.

If the video adapter compatibility is tied to the machine type, should it not select the most appropriate (or s known compatible) video adapter by default?  It's not intuitive to the end user, unless one has specific knowledge/experience of that particular motherboard's idiosyncrasies.
One would expect that if a default video adapter is indeed automatically selected when choosing a particular machine type or socket, then it would be fair to assume that video adapter is compatible with the machines within that socket / machine type category.

**Reply by eddmanx** [reply]
The default adapter for all machines that have ISA is "IBM CGA". If the board doesn't have ISA, IINM it defaults to "None".

Using 86box requires some knowledge of old PC hardware. Users are generally expected to pick and choose the different parts.

Video adapter compatibility is dependent on the mainboards' ROM, not the socket type or the CPU model. It's not something that is immediately apparent and requires testing. Trying to set different defaults for every single different mainboard would be quite time consuming.

**Comment by jimjamz** [comment]
The IBM PS/ValuePoint 433 DX/Si has only the ISA bus. I wrongly assumed the default-selected ISA IBM CGA to be compatible.  Before posting my question, I also tried the ISA Hercules, which is known to have wide compatibility, but it also doesn't work.  Therefore it appears the CGA category of video adapters are not compatible with some of the motherboard BIOS ROMs within the Socket 2 Machine type category.  Furthermore, I found the default IBM CGA video adapter to be incompatible with 3 other systems in the Socket 2 category:
- [ACC 2168] Packard Bell PB410A
- [ALi M1429G] Acer A1G
- [SIS 641] DEC DEC[c LPV

That would make more than half of the machines in that category not compatible with the default-selected video adapter.  
I know display adapter compatibility is not dependent on the socket type, but that is the parent category an end user first chooses from to refine their search for the system they want to build.
Forcing the end user to go on a compatible display hunt should they choose one of those systems seems rather unnecessary.

In the case of the J-Bond PCI400C-A (with its PCI bus: https://theretroweb.com/motherboards/s/j-bond-pci400c-a-pci-486), I tried selecting a PCI video adapter, which I anticipated to have generally wide compatibility across such systems, the Tseng Labs ET4000/w32 Rev. D, but the video adapter also failed.  Without exhaustively trying each video adapter against every single machine I'd attempted so far, the failures led me to believe that there may be issues with the ROMs of the tested machines, or most certainly with the default configuration.

I eventually found that the ISA IBM EGA video adapter worked, but I still can't explain why the PCI VGA card did not work.

I have working knowledge and experience with old PC hardware, particularly 486 and Pentium machines of that era, owning several.

I'd argue that the time spent implementing the selection of an appropriate default video adapter for each BIOS ROM (i.e Machine in 86Box) would be time well spent, especially for those users not familiar with which video display formats work with which ROMs (an ever-growing larger userbase now).

Many ROMs are similar and share commonalities based on their brand, chipset type or era (AMI, AWARD, etc.) and can therefore be categorised into BIOS ROM families.
It would be easier to implement a case switch or lookup table that has a ROM family associated to a default video adapter, whether that be CGA, EGA or VGA.  Sadly, I think the Machine:Video adapter relationship would be the only route.

As an example, both the Kaimer SA-486 VL-BUS M.B and the Acer A1G are both using the ALi M1429G BIOS, yet one works with CGA video adapters and the other does not.
